### PR TITLE
Add CRC checking support to consumer (#1056)

### PR DESCRIPTION
**Pull request by edenhill**

### CODE REVIEW

**Comment by ewencp** [comment]
@edenhill How about requesting reviews from @hqin and @mhowlett too?

File: src/rdkafka_broker.c [hunk 1]
```diff
@@ -3557,6 +3557,39 @@ rd_kafka_messageset_handle (rd_kafka_broker_t *rkb,
 		rd_kafka_buf_read_i64(rkbuf, &hdr.Offset);
 		rd_kafka_buf_read_i32(rkbuf, &hdr.MessageSize);
 		rd_kafka_buf_read_i32(rkbuf, &hdr.Crc);
+
+                if (rkb->rkb_rk->rk_conf.check_crcs &&
+                    !(hdr.Attributes & RD_KAFKA_MSG_ATTR_COMPRESSION_MASK)) {
+                        /* Verify CRC if desired.
+                         * Since compression algorithms have their own
+                         * checksums we dont botter checking CRCs of compressed
```
Comment (ewencp) [inline src/rdkafka_broker.c]: typo -- botter -> bother

**Comment by edenhill** [comment]
They are not yet on this repo so they can't be assigned as reviewers, but now they know what to do, right @hqin and @mhowlett? :)

**Comment by edenhill** [comment]
@ewencp Regarding tests: none of this code is easy to unit test unfortunately.
I couldn't find any tests for the java client either so I was thinking of adding a generic test to kafkatest.

File: src/rdkafka_broker.c (hunk 1 above)
Comment (edenhill) [inline src/rdkafka_broker.c]: that's bether

**Comment by mhowlett** [comment]
@ewencp very happy to review C code ( particularly written by @edenhill )

Logic looks good to me at face value. Will look deeper / verify through executing a test app if no LGTM by someone else in near future (just cautious 'cause this code is still new to me).

**Pull request merged**